src/main/java/com/ratifire/devrate/service/interview/InterviewHistoryService.java

```diff
@@ -0,0 +1,63 @@
+package com.ratifire.devrate.service.interview;
+
+import com.ratifire.devrate.dto.InterviewHistoryDto;
+import com.ratifire.devrate.entity.InterviewHistory;
+import com.ratifire.devrate.exception.InterviewHistoryNotFoundException;
+import com.ratifire.devrate.mapper.impl.InterviewHistoryMapper;
+import com.ratifire.devrate.repository.InterviewHistoryRepository;
+import com.ratifire.devrate.security.helper.UserContextProvider;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service for managing interview summaries.
+ */
+@Service
+@RequiredArgsConstructor
+public class InterviewHistoryService {
+
+  private final InterviewHistoryRepository interviewHistoryRepository;
+  private final UserContextProvider userContextProvider;
+  private final InterviewHistoryMapper interviewHistoryMapper;
+
+  /**
+   * Retrieves an InterviewSummary entity by its identifier.
+   *
+   * @param id the unique identifier of the InterviewSummary to be retrieved.
+   * @return the InterviewSummary associated with the provided id.
+   * @throws InterviewHistoryNotFoundException if no InterviewSummary is found for the given id.
+   */
+  public InterviewHistory findById(long id) {
+    return interviewHistoryRepository.findById(id).orElseThrow(() ->
+        new InterviewHistoryNotFoundException(id));
```

Comment:
we need method that `findByIdAndUser_Id` at the repository that retrieved only current user interview history object